src/m3ninx/index/regexp.go

```diff
@@ -62,7 +62,10 @@ func CompileRegex(r []byte) (CompiledRegex, error) {
 	if err != nil {
 		return CompiledRegex{}, err
 	}
-	compiledRegex := CompiledRegex{Simple: simpleRE}
+	compiledRegex := CompiledRegex{
+		Simple:       simpleRE,
+		VellumSyntax: vellumRe,
```

Comment:
nit: Would it be worth renaming this field to something like: "UnanchoredSyntax" if we're gonna start using it in other places?